client_base.go

```diff
@@ -228,6 +228,23 @@ type requestParameters struct {
 	contentType string
 }
 
+type ContextKey string
+
+func extractAdditionalHeaders(ctx context.Context) map[string]string {
+	additionalHeaders, ok := ctx.Value(ContextKey("additionalHeaders")).(map[string]string)
+	if ok {
+		// only take headers that start with Firebolt- prefix
+		filteredHeaders := make(map[string]string)
+		for key, value := range additionalHeaders {
+			if strings.HasPrefix(key, "Firebolt-") {
```

Comment:
Have we considered using `X-Firebolt-**` pattern?  `X` is a standard prefix for HTTP extension headers, and we are likely to end up with more custom headets set (see https://docs.google.com/document/d/1KqjIOrDLW4O1Y34pLr5oGa7f20tgc4Cx9Ae84oOMqQM/edit?tab=t.0#heading=h.ev90bjt1onfz)

Reply:
Oh, ok, I can update it then.

Reply:
@cosmpop FYI

Reply:
I'm not sure if this would require any update on packdb